config/config_mgmt.py

```diff
@@ -430,65 +478,84 @@ def mergeItems(it1, it2):
         return D1
 
     """
-    Create a defConfig for given Ports from Default Config File.
+    search Relevant Keys in Config using DFS, This function is mainly
+    used to search port related config in Default ConfigDbJson file.
+    In: Config to be searched
+    skeys: Keys to be searched in In Config i.e. search Keys.
+    Out: Contains the search result
     """
-    def getDefaultConfig(self, ports=list()):
+    def searchKeysInConfig(self, In, Out, skeys):
+
+        found = False
+        if isinstance(In, dict):
+            for key in In.keys():
+                #print("key:" + key)
+                for skey in skeys:
+                    # pattern is very specific to current primary keys in
+                    # config DB, may need to be updated later.
+                    pattern = '^' + skey + '\|' + '|' + skey + '$' + \
```

Comment:
why don't we prepare these patterns ahead of time using skeys and then search key in prepared patterns

Reply:
@vasant17 : any advantage of it?

One thing I want, is to keep these patterns at only one place, may be here or at start of library. So that if in future, configDB key pattern changes, we change at one place, except that Let me know if you see any other constraints.

Reply:
I was thinking of the same thing. If the skeys are static and we have superset, we can create all patterns only once through out the life of the program. Ans we can just update/change it if any new key gets introduced in config DB as we bring on new feature. 

If that is NOT the case and we want to prepare pattern for every skey in this method, I was trying to avoid creating/constructing pattern multiple times for the same skey. Hence I suggested to do it in the beginning and then iterate over keys to search in patterns.

Reply:
The idea is creating pattern once is good, but I will solve it soon after I finish DPB system tests so that I do not have a retest this entire critical part manually.